app/scripts/models/custom_cursor.js

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ * Q-MUNICATE custom cursor models Module
+ *
+ */
+define([], function() {
+    'use strict';
+
+    function Cursor(app) {
+        var self = this;
+
+        this.setCursorAfterElement = function(el) {
+            var range = document.createRange();
+
+            range.setStartAfter(el);
+            range.setEndAfter(el);
+
+            setRange(range);
+        };
+
+        this.setCursorToEnd = function(el) {
+            el.focus();
+            if (typeof window.getSelection !== 'undefined' &&
```

Comment:
Read this [article](http://prgssr.ru/development/15-sovetov-po-napisaniyu-samodokumentiruemogo-javascript.html) please,
You can create a variable `isRanfeAvaible`.